apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -145,19 +118,19 @@ export class UpsertWorkflowUseCase {
       active: isWorkflowActive,
       description: workflowDto.description || '',
       tags: workflowDto.tags || [],
-      userPreferences: command.workflowDto.preferences?.user ?? null,
-      defaultPreferences: command.workflowDto.preferences?.workflow ?? DEFAULT_WORKFLOW_PREFERENCES,
-      triggerIdentifier: command.preserveWorkflowId ? workflowDto.workflowId : slugify(workflowDto.name),
+      userPreferences: workflowDto.preferences?.user ?? null,
+      defaultPreferences: workflowDto.preferences?.workflow ?? DEFAULT_WORKFLOW_PREFERENCES,
+      triggerIdentifier: preserveWorkflowId ? workflowDto.workflowId : slugify(workflowDto.name),
       status: computeWorkflowStatus(isWorkflowActive, steps),
     };
   }
 
   private async buildUpdateWorkflowCommand(
-    workflowDto: UpsertWorkflowDataCommand,
-    user: UserSessionData,
+    command: UpsertWorkflowCommand,
     existingWorkflow: NotificationTemplateEntity
   ): Promise<UpdateWorkflowCommand> {
-    const steps = await this.mapSteps(workflowDto.origin, user, workflowDto.steps, existingWorkflow);
+    const { workflowDto, user } = command;
+    const steps = await this.buildSteps(command, existingWorkflow);
```

Comment:
sorry i dont get it. 
wont we create issues in here for the newly created workflow? (in this.buildSteps => this.buildStepIssuesUsecase.execute)